subprojects/groovy-macro/src/main/groovy/org/codehaus/groovy/macro/matcher/internal/MatchingConstraintsBuilder.groovy

```diff
@@ -24,7 +24,7 @@ import org.codehaus.groovy.macro.matcher.TreeContext
 import org.codehaus.groovy.syntax.Token
 
 class MatchingConstraintsBuilder {
-    Set<String> placeholders = new LinkedHashSet<>()
+    final Set<String> placeholders = new LinkedHashSet<>()
```

Comment:
Since this is a Groovy object this is changing a public property to final and I don't think should be done.